packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/ModelReaderWriterValidation/ModelTests.cs

```diff
@@ -40,6 +41,8 @@ protected virtual string GetExpectedResult(string format)
         }
         protected abstract void VerifyModel(T model, string format);
         protected abstract void CompareModels(T model, T model2, string format);
+        protected abstract Task TestBinaryContentCast(T model, ModelReaderWriterOptions options);
```

Comment:
I was thinking [something closer to this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/tests/public/ModelReaderWriterTests/RoundTripStrategy.cs#L167) so it can just be another wrapper test implemented for all models without needing to override these methods.